gtsam/slam/EssentialMatrixFactor.h

```diff
@@ -420,4 +417,104 @@ class EssentialMatrixFactor4
+ * calibration unknown, as Jacobians will be incorrect...
+ *
+ * Note: even stronger caveats about having priors on calibration apply.
+ */
+template <class CALIBRATION>
+class EssentialMatrixFactor5
+    : public NoiseModelFactorN<EssentialMatrix, CALIBRATION, CALIBRATION> {
+ private:
+  Point2 pA_, pB_;  ///< points in pixel coordinates
+
+  typedef NoiseModelFactorN<EssentialMatrix, CALIBRATION, CALIBRATION> Base;
+  typedef EssentialMatrixFactor5 This;
+
+  static constexpr int DimK = FixedDimension<CALIBRATION>::value;
+  typedef Eigen::Matrix<double, 2, DimK> JacobianCalibration;
+
+ public:
+  // Provide access to the Matrix& version of evaluateError:
+  using Base::evaluateError;
+
+  /**
+   *  Constructor
+   *  @param keyE Essential Matrix aEb variable key
+   *  @param keyKa Calibration variable key for camera A
+   *  @param keyKb Calibration variable key for camera B
+   *  @param pA point in first camera, in pixel coordinates
+   *  @param pB point in second camera, in pixel coordinates
+   *  @param model noise model is about dot product in ideal, homogeneous
+   * coordinates
+   */
+  EssentialMatrixFactor5(Key keyE, Key keyKa, Key keyKb, const Point2& pA,
+                         const Point2& pB, const SharedNoiseModel& model = nullptr)
+      : Base(model, keyE, keyKa, keyKb), pA_(pA), pB_(pB) {}
+
+  /// @return a deep copy of this factor
+  gtsam::NonlinearFactor::shared_ptr clone() const override {
+    return std::static_pointer_cast<gtsam::NonlinearFactor>(
+        gtsam::NonlinearFactor::shared_ptr(new This(*this)));
+  }
+
+  /// print
+  void print(
+      const std::string& s = "",
+      const KeyFormatter& keyFormatter = DefaultKeyFormatter) const override {
+    Base::print(s);
+    std::cout << "  EssentialMatrixFactor5 with measurements\n  ("
+              << pA_.transpose() << ")' and (" << pB_.transpose() << ")'"
+              << std::endl;
+  }
+
+  /**
+   * @brief Calculate the algebraic epipolar error pA' (Ka^-1)' E Kb pB.
+   *
+   * @param E essential matrix for key keyE
+   * @param Ka calibration for camera A for key keyKa
+   * @param Kb calibration for camera B for key keyKb
+   * @param H1 optional jacobian of error w.r.t E
+   * @param H2 optional jacobian of error w.r.t Ka
+   * @param H3 optional jacobian of error w.r.t Kb
+   * @return * Vector 1D vector of algebraic error
+   */
+  Vector evaluateError(const EssentialMatrix& E, const CALIBRATION& Ka,
+                       const CALIBRATION& Kb, OptionalMatrixType HE,
+                       OptionalMatrixType HKa,
+                       OptionalMatrixType HKb) const override {
+    // converting from pixel coordinates to normalized coordinates cA and cB
+    JacobianCalibration cA_H_Ka;  // dcA/dKa
+    JacobianCalibration cB_H_Kb;  // dcB/dKb
+    Point2 cA = Ka.calibrate(pA_, HKa ? &cA_H_Ka : 0, OptionalNone);
+    Point2 cB = Kb.calibrate(pB_, HKb ? &cB_H_Kb : 0, OptionalNone);
+
+    // convert to homogeneous coordinates
```

Comment:
nit: capitalize first letter (and on other lines)

Reply:
Done